tests/typescript-mock-data.spec.ts

```diff
@@ -51,36 +60,61 @@ it('should generate mock data functions with external types file import', async
     const result = await plugin(testSchema, [], { typesFile: './types/graphql.ts' });
 
     expect(result).toBeDefined();
-    expect(result).toContain("import { Avatar, UpdateUserInput, User, Status } from './types/graphql';");
+    expect(result).toContain(
+        "import { ABCType, Avatar, UpdateUserInput, User, ABCStatus, Status } from './types/graphql';",
+    );
     expect(result).toMatchSnapshot();
 });
 
 it('should generate mock data with typename if addTypename is true', async () => {
     const result = await plugin(testSchema, [], { typesFile: './types/graphql.ts', addTypename: true });
 
     expect(result).toBeDefined();
-    expect(result).toContain("import { Avatar, UpdateUserInput, User, Status } from './types/graphql';");
     expect(result).toMatchSnapshot();
 });
 
 it('should generate mock data with pascalCase enum if enumValues is "pascal-case#pascalCase"', async () => {
+    const result = await plugin(testSchema, [], { enumValues: 'pascal-case#pascalCase' });
+
+    expect(result).toBeDefined();
+    expect(result).toMatchSnapshot();
+});
+
+it('should generate mock data with upperCase enum if enumValues is "upper-case#upperCase"', async () => {
+    const result = await plugin(testSchema, [], { enumValues: 'upper-case#upperCase' });
+
+    expect(result).toBeDefined();
+    expect(result).toMatchSnapshot();
+});
+
+it('should generate mock data with as-is enum if enumValues is "keep"', async () => {
+    const result = await plugin(testSchema, [], { enumValues: 'keep' });
+
+    expect(result).toBeDefined();
+    expect(result).toMatchSnapshot();
+});
+
+it('should generate mock data with pascalCase types by default', async () => {
+    const result = await plugin(testSchema, [], {});
+
+    expect(result).toBeDefined();
+    expect(result).toMatchSnapshot();
```

Comment:
It's hard to see the expected value of all these 3 unit test without opening the snapshot.
Could you  add one more `expect` to check the typename only? This way, we'll directly see in the 5 lines of the test which option causes which output

Reply:
> Thanks for the PR ! I have a few comments but nothing major.
> Also you say in the description:
> 
> > Default casing of typenames to pascalCase for consistency with the typescript plugin - this is a breaking change
> Do you mean a breaking change on this library? Because it seems to be the same behaviour as before by default.

The current behaviour is in fact "keep", but it's easy to miss because 99% of typenames are already pascalCase to begin with. It is only noticeable if one has typenames either containing consecutive uppercase characters (such as `ABCType`), which isn't very common, or starting with a lowercase character, which is virtually non-existent.

So yes, the change of default behaviour will be a breaking change for a very small number of cases.